checker/bin/wpi-many.sh

```diff
@@ -191,15 +200,11 @@ do
 
     if [ "${OWNER}" = "${GITHUB_USER}" ]; then
         ORIGIN=$(echo "${REPOHASH}" | awk '{print $3}')
-        # Piping to /dev/null is usually a bad idea.
-        # However, in this case it is intentional: the goal is to
-        # suppress error output, because there is no real harm done if
-        # the `unannotated` remote is not added - it's just a convenience
-        # for data analysis. But, running this script twice in a row on projects
-        # whose owner is the github user always causes an error on this line,
-        # because the `unannotated` remote is already set. The output is piped
-        # to /dev/null to suppress that error.
-        git remote add unannotated "${ORIGIN}" &> /dev/null
+        # The `unannotated` remote is just a convenience for data analysis.
+        # Running this script twice in a row on projects whose owner
+        # is the github user always causes a harmless error on this
+        # line, because the `unannotated` remote is already set.
```

Comment:
Could you use `git ls-remote --exit-code ...` to test this?  That would be shorter than the comment explaining it...

Reply:
The advantage of the comment is that it is not executable, so it doesn't increase the script's complexity. I don't mind having a long comment in the script (which most users will never read). Using `|| true` here to capture the failure case seems fine to me, since it doesn't matter if this fails.

If you think this is serious issue, I'm happy to change it to a test using `git ls-remote` as you suggested, but I personally prefer it as-is.

Reply:
It's not a *serious* issue, but it's an issue.

The comment *does* increase the script's complexity:  it makes the script longer and harder to read and understand.

Using `|| true` is a code smell and might, in the future, obscure some other problem.  That is my concern.

What is the data analysis step?  Could this be moved to that script?

Reply:
> What is the data analysis step? Could this be moved to that script?

This feature was intended to allow the user to 1) run `wpi-many.sh` once on a set of repositories, 2) triage the results and write any necessary annotations in forks of the repositories, and 3) re-run the script using the newly-annotated forks while still being able to compute the correct lines of code and other statistics on the unannotated code.

I'm inclined to remove the feature entirely. It's complicated, and I've only ever used it for one experiment (the `DescribeImagesRequest` experiment in our ICSE 2020 paper on builders) - in more recent experiments, we haven't written any annotations by hand at all, and even if we did I'm not convinced this mechanism would be all that useful.

Reply:
That said, let's do that as a separate PR, since I'd like to merge this one now so that I can fix do-like-javac's CI.